### PR TITLE
Updated Documentation

### PR DESCRIPTION
**Pull request by Pyxus**

### CODE REVIEW

File: docs/getting_started/mod_collision/index.md [hunk 1]
```diff
@@ -2,74 +2,67 @@
 
 ## Introduction
 
-This module contains scripts related to hit detection and hitbox management.
+The Collision module provides tools for hit detection and hitbox management. In fighting games, hitbox configurations often change frequently. A character's hitboxes when standing will vary from the configuration when crouching, attacking, or getting hit. Different hitboxes can also detect different types of attacks and deal different amounts of damage. In Godot, managing multiple hitboxes with varying properties can be tedious, but the Collision module helps simplify the process.
 
 ## Hitbox2D/3D
 
-The hitbox node doesn't provide much functionality out of the box. But what it does provide is a template that you can expand upon through the use of attributes.
+The hitbox node is the simplest level of organization within the collision module. It doesn't provide much functionality out of the box and is essentially just an Area. But what it does provide is a template that you can expand upon through the use of attributes.
```
Comment (20milliliter) [inline docs/getting_started/mod_collision/index.md:9]: ``through the use of `HitboxAttributes`.``*

File: docs/getting_started/mod_state/index.md [hunk 2]
```diff
@@ -1,64 +1,50 @@
...
-
-
-## Adding CombatStateMachine node to the scene tree
-
-If the plugin is enabled then the CombatStateMachine should be available from the 'Create New Node' dialog.
-
-![Image of state machine node in create new node window](images/create_new_sm.png)
-
-![Image of state machine in scene tree](images/scene_tree.png)
+The state management module includes a hierarchical combat state machine for managing the state of combatants in your game, as well as a general-purpose state machine for capturing more general state flows. State management is an essential part of designing complex game  mehchanics as it helps define different states a game entity can be in and what actions can be performed in those states. In regards to fighting games that means defining what buttons transition into what attacks, and what attcks are available during any state.
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:5]: `complex game mehchanics, as it helps`*
`can be in, and what`*
`In regards to fighting games, that means `*

File: docs/getting_started/mod_state/index.md [hunk 3]
```diff
@@ -1,64 +1,50 @@
...
 
 ## Using the CombatStateMachine
 
-### Adding situations
+For this demonstration we will be using the `FrayCombatStateMachine` however all the information regarding building the states and transitions apply to the general purpose `FraySimpleStateMachine` as well. To get started first add a `FrayCombatStateMachine` from the node creation dialog.
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:9]: ``the `FrayCombatStateMachine`, however, all``*

File: docs/getting_started/mod_state/index.md [hunk 4]
```diff
@@ -1,64 +1,50 @@
...
 
-To add a new situation call the `add_situation()` method on the combat state machine. It accepts 2 arguments, a string name, and a `GraphNodeStateMachineGlobal`. 
+Once the state machine is added to the scene you can begin adding situations. A situation is a name to `FrayRootState` mapping which represents the set of actions available to a combatant. To add a new situation call the `add_situation()` method on the combat state machine. It accepts 2 arguments, a string name, and a `FrayRootState`. The root state is what actually contains all the information related to states and transitions.
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:11]: ``A situation is a `StringName` to `FrayRootState` mapping``*
``a `StringName`, and a `FrayRootState`.``*

File: docs/getting_started/mod_state/index.md [hunk 5]
```diff
@@ -1,64 +1,50 @@
...
+func _ready() -> void:
+    combat_state_machine.add_situation("on_ground", FrayRootState.builder()
+        .transition_button("idle", "attack_1", {input="btn_punch"})
+        .transition_sequence("idle", "attack_2", {sequence="seq_236p"})
+        .build("idle")
+    )
+```
 
-- The `build()` method will return a graph node built using the current build configuration. The first state added to the system will be used for the start state but you can optionally pass the start state to this method.
+- The `transition_button()` and `transition_sequence()` method both take 4 arguments. A 'from' state string, a 'to' state string, a string representing the input name, and an optional config dictionary. Note the input name is an arbitrary string name and does not inherently reference anything related to the input module.
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:27]: ``The `transition_button()` and `transition_sequence()` method both take 3 arguments. A 'from' `StringName`, a 'to' `StringName`, and an optional config `Dictionary`.``*

File: docs/getting_started/mod_state/index.md [hunk 6]
```diff
@@ -1,64 +1,50 @@
...
 The above situation could be visualized like this:
 
 ![Visulization of described situation](images/situation_visualization.svg)
 
 ### Transition configs
 
-The optional config dictioanry allows for additional custimization of the transition. A full list of availabe configs can be found in the doc comments located above the builder transition methods.
+The optional config dictionary allows for additional customization of the transition. Each key in the config dict corresponds to a property on the transition it uses.
 
 Below demonstates usage of the `prereqs` config. Prereqs take an array of conditions and only allow the transition to occur when all are true. To update the value of the condition the `set_condition()` method can be called on the `GraphNodeStateMachine`.
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:38]: `GraphNodeStateMachine` is an outdated name?

File: docs/getting_started/mod_state/index.md [hunk 7]
```diff
@@ -71,18 +57,18 @@ Global transitions are a convinience feature that allows you to automatically co
```
Comment (20milliliter) [inline docs/getting_started/mod_state/index.md:57]: `If a rule exists, states`*
`tagged as 'special'.`*

File: docs/getting_started/index.md [hunk 8]
```diff
@@ -1,94 +1,11 @@
...
 
 ## Modules
 
 - [Collision](./mod_collision/index.md)
 
 - [Input](./mod_input/index.md)
 
-- [State Management](./mod_state/index.md)
-
-## Quickstart
```
Comment (20milliliter) [inline docs/getting_started/index.md]: Was this intentional?

File: docs/getting_started/mod_input/index.md [hunk 9]
```diff
@@ -2,63 +2,93 @@
...
-
-All builder methods returns a reference to the builder itself to allow for optional method chaining.
-
-Example Usage:
+### Registering Composites
 
-```gdscript
-const CIF = Fray.Input.CompositeInputFactory
-const CombinationMode = Fray.Input.CombinationInput.Mode
+Composite inputs can be registered in a similar manner to binds using the `FrayInputMap.add_composite_input()` method. The method takes two arguments a name for the composite which must be unique between both composites and binds, and a composite object which can be built using the included builder classes. Fray comes with three composite inputs: Combinations, Conditionals, Groups, and Simple. Simple inputs are effectively just wrappers around binds and are used as the leafs of composition.
```
[the remark this answers is not in the document]
Comment (Pyxus) [reply docs/getting_started/mod_input/index.md:28]: I'm so used to working on this solo I wasn't expecting any reviewers. I'll need to get into the habit of making more atomic commits if this keeps ups 🙂. Thanks for taking the time to read through the changes, I'll be addressing them all when I have time. And yes this was a typo, coupled with a leftover explanation about simple inputs.

[the remark this answers is not in the document]
Comment (20milliliter) [reply docs/getting_started/mod_input/index.md:28]: It's the least I can do. 👍

[the remark this answers is not in the document]
Comment (20milliliter) [reply docs/getting_started/mod_input/index.md:28]: _Actually_, making a PR is the least I can do. #37

File: docs/getting_started/index.md [hunk 10]
```diff
@@ -1,13 +1,14 @@
...
-- [State Management](./mod_state/index.md)
+- [Collision](./mod_collision/index.md)
 
 - [Input](./mod_input/index.md)
 
-- [Collision](./mod_collision/index.md)
-
+- [State Management](./mod_state/index.md)
 
 ## Quickstart
```
[the remark this answers is not in the document]
Comment (Pyxus) [reply docs/getting_started/index.md]: Yes. I was debating how to approach the "getting started" rewrite at the time and decided thoroughly explaining how to use the main features in sections with code examples would be more useful than a large commented example. It was less removed and more split up. I'm open to hearing arguments in favor of keeping it (perhaps extending to include features from other modules, or maybe giving each module a quickstart).

**Pull request merged**